src/lib.rs

```diff
@@ -56,6 +61,74 @@ impl Reading {
     }
 }
 
+#[derive(Debug)]
+struct Output {
```

Comment:
One idea for you - what if we called this `Log`, or `Logger` or even more specific `ResponseLog` \ `ResponseLogger` instead of `Output`? There may even be naming convention that's common in other Rust programs here (and perhaps that is `Output` and I just aren't aware of this).